crates/state/api/src/lib.rs

```diff
@@ -79,7 +79,7 @@ pub trait StateDebug {
     type Error;
 
     /// Retrieves the storage root of the account at the specified address.
-    fn account_storage_root(&self, address: &Address) -> Result<Option<B256>, Self::Error>;
+    fn account_storage_root(&self, address: &Address) -> Option<B256>;
```

Comment:
Issue: This is a general-purpose API intended for extensibility by other `State` implementations. We cannot control whether they will or will not return an error, so we should keep this API returning a Result for future-compatibility; even if our current internal implementations don't return an error.